### PR TITLE
Use AnnotationSlim for the queries of the bulk annotation API

### PR DESCRIPTION
**Pull request by marcospri**
~~Requires: https://github.com/hypothesis/h/pull/8260~~

~~(This could been just one PR, I thought this would be more complex)~~

Changed the approach for the factories, just calling the Slim factory on these tests.

## Testing

Follow the standard email generation procedure, eg: https://github.com/hypothesis/lms/pull/5694

### CODE REVIEW

File: h/services/bulk_annotation.py [hunk 1]
```diff
@@ -109,36 +109,34 @@ def annotation_search(
...
         """Generate a query which can then be executed to find annotations."""
-        query = sa.select(
-            [cls._AUTHOR.username, Group.authority_provided_id]
-        ).select_from(Annotation)
-
+        query = (
+            sa.select([cls._AUTHOR.username, Group.authority_provided_id]).select_from(
+                AnnotationSlim
+            )
+            # Always join on AUTHOR as we return its username
```
Comment (marcospri) [inline h/services/bulk_annotation.py:116]: Join while creating the query.

This is probably a matter of taste but I prefer this order, first build the query with all necessary joins, then just add filters.

File: h/services/bulk_annotation.py [hunk 2]
```diff
@@ -109,36 +109,34 @@ def annotation_search(
...
+            .join(cls._AUTHOR, cls._AUTHOR.id == AnnotationSlim.user_id)
+            # Always join on Group as we return the authority_provided_id
+            .join(Group, Group.id == AnnotationSlim.group_id)
+        )
         # Updated
-        query = query.where(date_match(Annotation.updated, updated))
+        query = query.where(date_match(AnnotationSlim.updated, updated))
```
Comment (marcospri) [inline h/services/bulk_annotation.py:122]: These just use the same column in the new table.

File: h/services/bulk_annotation.py [hunk 3]
```diff
@@ -109,36 +109,34 @@ def annotation_search(
...
-        query = query.where(Annotation.shared.is_(True))
+        query = query.where(AnnotationSlim.shared.is_(True))
 
         # Deleted
-        query = query.where(Annotation.deleted.is_(False))
+        query = query.where(AnnotationSlim.deleted.is_(False))
 
         # Audience
-        query = query.join(Group, Annotation.groupid == Group.pubid).where(
+        query = query.where(
```
Comment (marcospri) [inline h/services/bulk_annotation.py:131]: This stays the same except for the join, that know happens on the FK.

File: h/services/bulk_annotation.py [hunk 4]
```diff
@@ -109,36 +109,34 @@ def annotation_search(
...
         )
 
         # NIPSA
-        query = query.join(
-            cls._AUTHOR,
-            cls._AUTHOR.username
-            == sa.func.split_part(sa.func.split_part(Annotation.userid, "@", 1), ":", 2)
-            and cls._AUTHOR.authority == authority,
-        ).where(cls._AUTHOR.nipsa.is_(False))
+        query = query.where(cls._AUTHOR.nipsa.is_(False))
```
Comment (marcospri) [inline h/services/bulk_annotation.py:136]: Join is simpler, using the FK above, the filter on author is the same.

File: h/services/bulk_annotation.py [hunk 5]
```diff
@@ -109,36 +109,34 @@ def annotation_search(
...
 
         # Moderated
-        query = query.outerjoin(AnnotationModeration).where(
-            AnnotationModeration.id.is_(None)
-        )
+        query = query.where(AnnotationSlim.moderated.is_(False))
```
Comment (marcospri) [inline h/services/bulk_annotation.py:139]: This is now part of the table, no join required.

File: h/services/bulk_annotation.py (hunk 2 above)
Comment (seanh) [reply h/services/bulk_annotation.py:122]: :+1:

Comment (seanh) [reply h/services/bulk_annotation.py:122]: Also already exists on `main` but I think you can do:

```python
query = (
    sa.select(...)
    .join(...)
    .join(...)
    .where(
        date_match(AnnotationSlim.updated, updated),
        AnnotationSlim.shared.is_(True),
        AnnotationSlim.deleted.is_(False),
        ...
    )
)
```

You don't need to do this thing where it goes:

```python
query = sa.select(...).join(...).join(...)

query = query.where(...)

query = query.where(...)

query = query.where(...)
```

**Review by seanh — APPROVED** [review]

**Pull request merged**